router/router.go

```diff
@@ -0,0 +1,188 @@
+package router
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/coreos/go-etcd/etcd"
+
+	"github.com/spesnova/iruka/schema"
+)
+
+const (
+	DefaultKeyPrefix = "/vulcand"
+	DefaultMachines  = "http://localhost:4001"
+)
+
+type Router struct {
+	etcd      etcd.Client
+	keyPrefix string
+}
+
+func NewRouter(machines, keyPrefix string) *Router {
+	if os.Getenv("IRUKA_ETCD_MACHINES") != "" {
+		machines = os.Getenv("IRUKA_ETCD_MACHINES")
+	}
+
+	if machines == "" {
+		machines = DefaultMachines
+	}
+
+	m := strings.Split(machines, ",")
+	etcdClient := *etcd.NewClient(m)
+	return &Router{etcdClient, keyPrefix}
+}
+
+// TODO(dtan4):
+//   Divide by upstream (blue/green)
+func (r *Router) AddBackend(name string) error {
+	backend := schema.VulcandBackend{
+		Type: "http",
+	}
+	j, err := marshal(backend)
+
+	if err != nil {
+		return err
+	}
+
+	key := path.Join(r.keyPrefix, "backends", name, "backend")
+	_, err = r.etcd.Create(key, string(j), 0)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Router) RemoveBackend(name string) error {
+	key := path.Join(r.keyPrefix, "backends", name, "backend")
+	_, err := r.etcd.Delete(key, true)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TODO(dtan4):
+//   Divide by upstream (blue/green)
+func (r *Router) AddServer(appID, containerName, url string) error {
```

Comment:
I think you don't need to pass 3 parameters (`appID`, `containerName`, `url`).
Pass `container schema.Container` to `AddServer` instead of those parameters.
Because `container` has those values:

* `appID` -> `container.AppID`
* `contanerName` -> `container.Name`
* `url` -> `"http://" + c.Machine + ":" + strconv.FormatInt(c.PublishedPort, 10)`

See: [Container Attributes - Iruka API References](https://github.com/spesnova/iruka/blob/master/docs/api-v1-alpha.md#attributes-2)